lib/spack/spack/main.py

```diff
@@ -744,8 +744,7 @@ def main(argv=None):
     # activate an environment if one was specified on the command line
     if not args.no_env:
         env = spack.cmd.find_environment(args)
-        if env:
-            ev.activate(env, args.use_env_repo, add_view=False)
+        ev.activate(env, args.use_env_repo)
```

Comment:
Not sure what to do here. The fact that `activate` now accepts a `None` is good, since it makes the method symmetric with deactivate. On the caller side though it was more readable before, since now a reader needs to know that:
1. `env` can be `None`
2. A `None` argument to `activate` is a no-op

Can we update the comment above and make to document this behavior at the calling site?

Reply:
In that case it's maybe best to just undo the activate-accepting-none-changes

Reply:
Or we can add an additional Boolean argument to activate/deactivate to accept or not `None`.

Reply:
Done that